### PR TITLE
Migrated social example from learn

### PR DESCRIPTION
**Pull request by 0xaptosj**
[Original social example demo using nextjs + chakra](https://aptos-move-friends.vercel.app/)

migrated version demo
https://github.com/user-attachments/assets/44cfd51b-e2d5-4710-b9ff-9a4ad8ef247c

### CODE REVIEW

File: examples/aptos-friend/frontend/pages/MyProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { useHasIssuedShare } from "@/hooks/useIssuer";
+import { IssuerDetails } from "@/components/IssuerDetails";
+import { IssuerShareHolders } from "@/components/IssuerShareHolders";
+import { IssueShare } from "@/components/IssueShare";
+import { UserHoldings } from "@/components/UserHoldings";
+import { TradeShare } from "@/components/TradeShare";
+
+export function MyProfile() {
+  const { connected, account } = useWallet();
+  let hasIssuedShare = useHasIssuedShare(account?.address as `0x${string}`);
```
Comment (0xmaayan) [inline examples/aptos-friend/frontend/pages/MyProfile.tsx:14]:
```suggestion
  const hasIssuedShare = useHasIssuedShare(account?.address as `0x${string}`);
```

File: examples/aptos-friend/frontend/components/UserHoldings.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Table, TableBody, TableCaption, TableCell, TableHead, TableHeader, TableRow } from "@/components/ui/table";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+import { useGetHoldings } from "@/hooks/useHolding";
+
+type UserHoldingsProps = {
+  userAddress: `0x${string}`;
+};
+
+export function UserHoldings({ userAddress }: UserHoldingsProps) {
+  let holdings = useGetHoldings(userAddress);
```
Comment (0xmaayan) [inline examples/aptos-friend/frontend/components/UserHoldings.tsx:12]:
```suggestion
  const holdings = useGetHoldings(userAddress);
```

File: examples/aptos-friend/frontend/components/TradeShare.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  const [amount, setAmount] = useState(0);
+  const [isBuying, setIsBuying] = useState(true);
+
+  const handleTradeSubmit = async () => {
+    if (!walletClient || !issuerObjectAddress) return;
+
+    let resp;
+    if (isBuying) {
+      resp = await walletClient.useABI(ABI).buy_share({
+        type_arguments: [],
```
Comment (0xmaayan) [inline examples/aptos-friend/frontend/components/TradeShare.tsx:35]: i think you should be able to not include it if it is empty
```suggestion
       
```

File: examples/aptos-friend/frontend/components/TradeShare.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+type TradeShareProps = {
+  issuerAddress: `0x${string}`;
+};
+
+export function TradeShare({ issuerAddress }: TradeShareProps) {
+  const { client: walletClient } = useWalletClient();
+  const { account } = useWallet();
+  const { toast } = useToast();
+  const holding = useGetHolding(issuerAddress, account?.address as `0x${string}`);
```
Comment (0xmaayan) [inline examples/aptos-friend/frontend/components/TradeShare.tsx:23]: why not just changing the holder account address type to `string`?

File: examples/aptos-friend/frontend/components/IssuerShareHolders.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import { useGetHolders } from "@/hooks/useHolder";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "@/components/ui/table";
+
+type IssuerShareHoldersProps = {
+  issuerAddress: `0x${string}`;
+};
+
+export function IssuerShareHolders({ issuerAddress }: IssuerShareHoldersProps) {
+  let holders = useGetHolders(issuerAddress);
```
Comment (0xmaayan) [inline examples/aptos-friend/frontend/components/IssuerShareHolders.tsx:12]:
```suggestion
  const holders = useGetHolders(issuerAddress);
```

File: examples/aptos-friend/frontend/components/IssuerDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { LabelValueGrid } from "@/components/LabelValueGrid";
+import { useGetIssuer } from "@/hooks/useIssuer";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+
+type IssuerDetailsProps = {
+  issuerAddress: `0x${string}`;
+};
+
+export function IssuerDetails({ issuerAddress }: IssuerDetailsProps) {
+  let issuer = useGetIssuer(issuerAddress);
```
Comment (0xmaayan) [inline examples/aptos-friend/frontend/components/IssuerDetails.tsx:10]:
```suggestion
  const issuer = useGetIssuer(issuerAddress);
```

File: examples/aptos-friend/frontend/utils/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Network } from "@aptos-labs/ts-sdk";
+import { NetworkInfo, isAptosNetwork } from "@aptos-labs/wallet-adapter-react";
+
+export const isValidNetworkName = (network: NetworkInfo | null) => {
```
Comment (0xmaayan) [inline examples/aptos-friend/frontend/utils/helpers.ts:4]: doesnt seem like this function is being used?

File: examples/aptos-friend/move/sources/aptos_friend.move [hunk 8]
```diff
@@ -0,0 +1,707 @@
...
+            extend_ref: object::generate_extend_ref(vault_constructor_ref),
+        });
+        move_to(sender, IssuerRegistry {
+            issuers: vector::empty()
+        });
+    }
+
+    // ================================= Entry Functions ================================= //
+
+    public entry fun issue_share(
```
Comment (0xmaayan) [inline examples/aptos-friend/move/sources/aptos_friend.move:107]: lets add doc string for each entry/view function

**Review by 0xmaayan — APPROVED** [review]
Left some comments, overall LGTM

File: examples/aptos-friend/frontend/components/TradeShare.tsx (hunk 3 above)
Comment (0xaptosj) [reply examples/aptos-friend/frontend/components/TradeShare.tsx:35]: this is surf's requirement

File: examples/aptos-friend/frontend/components/TradeShare.tsx (hunk 4 above)
Comment (0xaptosj) [reply examples/aptos-friend/frontend/components/TradeShare.tsx:23]: surf requires all address to be `0x${string}`

**Pull request merged**